File: Review-04-Unit-Testing/Example-7/tests/examples/test_game.py
from hamcrest import *
import unittest

from examples import (Player, Referee, Board, Game)

import copy


class TestGame(unittest.TestCase):
    """
    1 - Does this piece of code perform the operations
        it was designed to perform?

    2 - Does this piece of code do something it was not
        designed to perform?

    1 Test per mutator
    """
    def setUp(self):

        self.tom = Player("Tom")
        self.a_cylon = Player()

        self.empty_board = Board()

        self.a_game = Game(self.tom, self.a_cylon)

    def test_constructor(self):

        assert_that(self.a_game.get_player1(), equal_to(self.tom))
        assert_that(self.a_game.get_player2(), equal_to(self.a_cylon))

        assert_that(self.a_game.get_player1().get_symbol(), is_('X'))
        assert_that(self.a_game.get_player2().get_symbol(), is_('O'))

        assert_that(self.a_game.is_over(), is_(False))

        assert_that(self.a_game.get_winner(), is_(none()))
        assert_that(self.a_game.get_loser(), is_(none()))

        # Can not test without Board.equals method
        assert_that(self.a_game.get_board(), equal_to(self.empty_board))

    @unittest.skip("can not test")
    def test_play_round():

        # Can not test due to hardcoded System.in use in Player.next_move
        pass

File: 15-Java-Strategy-Pattern/Example-2/src/test/java/edu/odu/cs/cs330/examples/TestGame.java
package edu.odu.cs.cs330.examples;

import org.junit.jupiter.api.TestMethodOrder;
import org.junit.jupiter.api.MethodOrderer;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeEach;

import static org.junit.jupiter.api.Assertions.*;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.*;
// import org.hamcrest.core.IsNull;

/**
 * 1 - Does this piece of code perform the operations
 *     it was designed to perform?
 *
 * 2 - Does this piece of code do something it was not
 *     designed to perform?
 *
 * 1 Test per mutator
 */
@TestMethodOrder(MethodOrderer.MethodName.class)
public class TestGame
{
    Player tom;
    Player aCylon;

    Board emptyBoard;

    Game aGame;

    @BeforeEach
    public void setUp()
    {
        tom       = new Player("Tom", null);
        aCylon    = Player.REFERENCE_CYLON.clone();

        emptyBoard = new Board();

        aGame = Game.builder()
            .withPlayer(tom)
            .withPlayer(aCylon)
            .build();
    }

    @Test
    public void testConstructor()
    {
        assertThat(aGame.getPlayer1(), equalTo(tom));
        assertThat(aGame.getPlayer2(), equalTo(aCylon));

        assertThat(aGame.isOver(), is(false));

        assertThat(aGame.getWinner(), is(nullValue()));
        assertThat(aGame.getLoser(), is(nullValue()));

        // Can not test without Board.equals method
        assertThat(aGame.getBoard(), equalTo(emptyBoard));
    }

    @Test
    public void testPlayMatch()
    {
        fail("Not yet implemented");
    }

    @Test
    public void testPlayRound()
    {
        fail("Not yet implemented");
    }


    @Test
    public void testPerformTurn()
    {
        fail("Not yet implemented");
    }
}
